lib/cldr/language.ex

```diff
@@ -134,20 +134,28 @@ defmodule Cldr.Language do
         end
 
         # Error cases for available_locales/known_locales
-        def available_languages(locale), do: {:error, Locale.locale_error(locale)}
+        def available_languages(locale) when is_binary(locale) do
+          with {:ok, locale} <- backend().validate_locale(locale) do
+            available_languages(locale)
+          end
+        end
 
-        def known_languages(locale), do: {:error, Locale.locale_error(locale)}
+        def known_languages(locale) when is_binary(locale) do
+          with {:ok, locale} <- backend().validate_locale(locale) do
+            known_languages(locale)
+          end
+        end
```

Comment:
thanks for the comments - spot on as always. Fixed and pushed another commit.

Reply:
The errors are back, but they'll never be called. The branch before them will be recurse to infinity.

Reply:
The guarantees are this:
1. validate_locale/1 will only return an :ok tuple if the locale is both valid and known
2. The recursive call to known_languages/1 will therefore only happen if we know the call will be satisfied by one of the generated clauses.

So I don't believe it will recurse infinitely but I'll add a test to demonstrate that.

Reply:
I've pushed another commit with a test to ensure that infinite recursion does not occur because of the contract of `validate_locales/1`.